File: polyglot/windows/go/seis_windows_platform.go
package windowsplatform

type Capability struct {
	ID            string
	Languages     []string
	QualityGates  []string
	OfflineHelper bool
	RemoteBridge  bool
}

func CapabilitySurface() Capability {
	return Capability{
		ID:            "windows-go-service",
		Languages:     []string{"Go", "PowerShell", "SQL", "Python"},
		QualityGates:  []string{"go_build_when_available", "windows_path_safety", "permission_scope", "offline_fallback", "event_log_awareness"},
		OfflineHelper: true,
		RemoteBridge:  true,
	}
}

func IsReadyForSEISAgent(capability Capability) bool {
	return len(capability.Languages) >= 4 &&
		len(capability.QualityGates) >= 5 &&
		capability.OfflineHelper &&
		capability.RemoteBridge
}

File: polyglot/windows/rust/seis_windows_platform.rs
pub struct Capability {
    pub id: &'static str,
    pub languages: &'static [&'static str],
    pub quality_gates: &'static [&'static str],
    pub offline_helper: bool,
    pub remote_bridge: bool,
}

pub fn capability_surface() -> Capability {
    Capability {
        id: "windows-rust-core",
        languages: &["Rust", "PowerShell", "C++", "Go", "SQL", "Java"],
        quality_gates: &[
            "rustc_check_when_available",
            "windows_path_safety",
            "permission_scope",
            "offline_fallback",
            "event_log_awareness",
        ],
        offline_helper: true,
        remote_bridge: true,
    }
}

pub fn is_ready_for_seis_agent(capability: &Capability) -> bool {
    capability.languages.len() >= 4
        && capability.quality_gates.len() >= 5
        && capability.offline_helper
        && capability.remote_bridge
}
